connection_manager.go

```diff
@@ -322,8 +246,24 @@ func (n *connectionManager) handleInvalidCertificate(now time.Time, hostinfo *Ho
 	// Inform the remote and close the tunnel locally
 	n.intf.sendCloseTunnel(hostinfo)
 	n.intf.closeTunnel(hostinfo)
-
-	n.ClearLocalIndex(hostinfo.localIndexId)
-	n.ClearPendingDeletion(hostinfo.localIndexId)
+	delete(n.pendingDeletion, hostinfo.localIndexId)
 	return true
 }
+
+func (n *connectionManager) sendPunch(hostinfo *HostInfo) {
+	if hostinfo.remote == nil || !n.punchy.GetPunch() {
+		// Don't punch to relayed hosts
```

Comment:
should GetTargetEverything spam the network with 1-byte packets for every advertised remote address? If so, then it should probably spam all those addrs even for hosts currently going through a relay.

Especially if the purpose of it is to prime the network (host firewalls, etc.) for a tunnel migration.